src/code_8001F980.c

```diff
@@ -1372,7 +1372,11 @@ void func_800248D0(Player *player, s8 arg1, s8 arg2, s8 arg3) {
             sp154[2] = player->pos[2] + sp140;
         }
     }
-    D_80164B04 = &D_802F1F80[D_801651D0[arg2][arg1]][arg2][arg1 << 7];
+#ifdef AVOID_UB
+    D_80164B04 = &D_802F1F80[D_801651D0[arg2][arg1]][arg2][arg1].kart_palette;
```

Comment:
Not that this really matters, but I don't think this function is loading just the `kart_palette`. I think it wants the address of the entire `D_802F1F80` entry given its use in the `gDPLoadTLUT_pal256` call. Granted those two things will have the same address in practice, so this is mostly a pedantic nitpick.

Reply:
I though the name "kart" was in general since it was just there, but i guess it would make sense if both kart and wheel use the same tmem memory, i still find it weird how they use a different offset for the wheel pallete though.

Reply:
Oh, there's an entire thing going on here. I will try to summarize it (my summaries are usually bad, so feel free to ask for clarifications)

Animation frames for the characters are CI8 (palette using) images. But the palettes used by those images are unique. Instead of a palette per frame, there's a so-called "kart palette" that takes up the upper 3/4's of the palettes color space. The bottom 1/4 is reserved for the "wheel palette". There is an obvious misnomer here: aren't the wheels part of the kart? In this explanation, no.

The entries in `D_802F1F80` are full palettes, but they have component parts, the kart and wheel parts. The way the game animates the wheels (to give the illusion of them rolling) is to rapidly cycle the wheel part of the palette through a set of 1/4 sized palettes (4 per animation frame). That's what `func_80026B4C` is responsible for, choosing which "wheel palette" to use.

Reply:
What name would you suggest instead of `kart_palette`?

Reply:
The name `kart_palette` is fine, what I was originally getting at was that `&D_802F1F80[D_801651D0[arg2][arg1]][arg2][arg1].kart_palette` would be better replaced with `&D_802F1F80[D_801651D0[arg2][arg1]][arg2][arg1]` since that's """more correct""". But, as originally mentioned, that's a pedantic nitpick, its doesn't actually matter. It just bothers my aesthetic sensibilities.